receiver/receivercreator/observerhandler.go

```diff
@@ -67,9 +67,18 @@ func (obs *observerHandler) OnAdd(added []observer.Endpoint) {
 
 	for _, e := range added {
 		for _, template := range obs.receiverTemplates {
-			if !ruleMatches(template.Rule, e) {
+			if matches, err := template.rule.eval(e); err != nil {
+				obs.logger.Error("failed matching rule", zap.String("rule", template.Rule), zap.Error(err))
```

Comment:
Can this happen frequently? If so it may flood the logs. Consider using a metric instead or use `Debug` log level: https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#logging

Reply:
Possibly. It'll happen once per discovered pod per invalid rule. For example if a user writes a rule that uses a variable that doesn't exist it'll be logged per-pod. However it'll only logged be the first time it's observed. The same pod/rule won't be continuously logged. Do you think that's still too much?

Reply:
I think it is fine if it is once after the pod is observed. Let's keep it.